flink-runtime/src/main/java/org/apache/flink/runtime/execution/librarycache/BlobServerLibraryManager.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.execution.librarycache;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.blob.BlobServer;
+
+import javax.annotation.Nonnull;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Specialisation of {@link BlobLibraryCacheManager} that works with a {@link BlobServer}.
+ */
+public final class BlobServerLibraryManager extends BlobLibraryCacheManager {
```

Comment:
Oh yes, the API is not too clean here - let my give it another shot: I wanted to get rid of the `BlobService#registerJob()` and `BlobService#releaseJob()` calls and only want to make them available in the `BlobCache`. Let's extract these two calls from the `BlobLibraryCacheManager` since they should be called with the `FallbackLibraryCacheManager` as well...